decidim-surveys/app/commands/decidim/surveys/admin/publish_survey.rb

```diff
@@ -27,7 +27,7 @@ def call
 
           transaction do
             publish_survey
-            delete_answers unless @questionnaire.published?
+            delete_answers if @survey.clean_after_publish?
```

Comment:
We need to check this in a test, can you point me to which test is covering this?

Reply:
decidim-surveys/spec/system/admin_manages_surveys_spec.rb:92